openedx/features/colaraz_features/filters.py

```diff
@@ -0,0 +1,88 @@
+"""
+Filters for Colaraz Views.
+"""
+from opaque_keys.edx.django.models import CourseKeyField
+from student.roles import CourseInstructorRole
+from student.models import CourseAccessRole
+
+
+class CourseAccessRoleFilterMixin(object):
+    """
+    Filter to queryset based on user's course access roles.
+
+    This Mixin can be used on class based views with `get_queryset` method.
+    It applies filters based on the requesting user so that only authorized users are able to view/update/delete.
+    """
+
+    def get_queryset(self):
+        """
+        Override `get_queryset` to apply course access role based filtering.
+        """
+        queryset = super(CourseAccessRoleFilterMixin, self).get_queryset()
+        user = self.request.user
+
+        if user.is_staff:
+            return self.apply_global_staff_filter(queryset)
+
+        try:
+            instructor_role = self.get_organizational_role(user, CourseInstructorRole.ROLE)
+        except CourseAccessRole.DoesNotExist:
+            # Let it pass if user is not an organizational instructor.
+            pass
+        else:
+            return self.apply_organization_instructor_filter(queryset, instructor_role.org)
+
+        # Empty queryset for all other users
+        return queryset.none()
```

Comment:
Can we write this statement in the except block above?

Reply:
IMHO, That will reduce code readability. More filters will be added later on here, so it will eventually have to be moved down again.